### ISSUE TITLE
Cannot use this plugin when i dont have access to CREATE s3 bucket

### ISSUE THREAD

**Issue by jibin-tj** [jenkinsci/aws-beanstalk-publisher-plugin#27]
My IAM policy restricts creating s3 buckets,but i can put objects in existing buckets. When trying to deploy the code, i get a the following error
`com.amazonaws.services.elasticbeanstalk.model.InsufficientPrivilegesException:You do not have permission to perform the 's3:CreateBucket' action. Verify that your S3 policies and your ACLs allow you to perform these actions. (Service: AWSElasticBeanstalk; Status Code: 403; Error Code: InsufficientPrivilegesException; Request ID: 175c4f6d-0c3e-11e6-bb6f-e18e70ceaeaa) at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:1307) at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:894) at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:597) at com.amazonaws.http.AmazonHttpClient.doExecute(AmazonHttpClient.java:363) at com.amazonaws.http.AmazonHttpClient.executeWithTimer(AmazonHttpClient.java:329) at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:308) at com.amazonaws.services.elasticbeanstalk.AWSElasticBeanstalkClient.invoke(AWSElasticBeanstalkClient.java:2039) at com.amazonaws.services.elasticbeanstalk.AWSElasticBeanstalkClient.updateEnvironment(AWSElasticBeanstalkClient.java:1924) at org.jenkinsci.plugins.awsbeanstalkpublisher.AWSEBEnvironmentUpdaterThread.updateEnv(AWSEBEnvironmentUpdaterThread.java:82) at org.jenkinsci.plugins.awsbeanstalkpublisher.AWSEBEnvironmentUpdaterThread.run(AWSEBEnvironmentUpdaterThread.java:215) at org.jenkinsci.plugins.awsbeanstalkpublisher.AWSEBEnvironmentUpdaterThread.call(AWSEBEnvironmentUpdaterThread.java:185) at org.jenkinsci.plugins.awsbeanstalkpublisher.AWSEBEnvironmentUpdaterThread.call(AWSEBEnvironmentUpdaterThread.java:21) at java.util.concurrent.FutureTask.run(FutureTask.java:262) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) at java.lang.Thread.run(Thread.java:745)`

Does this error occurs because the plugin tries to insert the code to some temporary s3 bucket.
In short ,i need to use this plugin even though i don't have createbucket privilege

**Comment by DavidTanner** [comment]
I am not sure how to address this, I have never seen this error before.  As far as I know there is nothing in the plugin that tries to create a new bucket.  can you post the relevant portion of the config?

**Comment by DavidTanner** [comment]
It may be because you don't already have a bucket in S3, so they try to do it for you.  Make sure the bucket already exists before the job runs.
